File: src/Easy/Leetcode283.java
package Easy;

public class Leetcode283 {
    public void moveZeroes(int[] nums) {
        int numZeroes=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                numZeroes++;
            }
            else{
                nums[i-numZeroes]=nums[i];
            }
        }
        int index=nums.length-1;
        while(numZeroes>0){
            nums[index]=0;
            numZeroes--;
            index--;
        }
    }
}

File: src/Python/Easy/Leetcode283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zero_count=0
        for i,num in enumerate(nums):

            if num==0:
                zero_count+=1
            else:
                nums[i-zero_count]=nums[i]



        for i in range(zero_count):
            nums[len(nums)-1-i]=0
